gulpfile.js

```diff
@@ -246,21 +246,21 @@ gulp.task('ext:test', (done) => {
 	var args = ['--verbose', '--disable-gpu', '--disable-telemetry', '--disable-updates', '-n'];
 	let vscodeVersion = packageJson.engines.vscode.slice(1);
 	let extensionTestsPath = `${workspace}/out/test`;
-	vscodeTest.downloadAndUnzipVSCode(vscodeVersion).then((vscodePath) => {
-		if (vscodePath) {
-			vscodeTest.runTests({
-				vscodeExecutablePath: vscodePath,
-				extensionDevelopmentPath: workspace,
-				extensionTestsPath: extensionTestsPath,
-				launchArgs: args
-			}).then(() => done()).catch((error) => {
-				console.log(`stdout: ${process.stdout}`);
-				console.log(`stderr: ${process.stderr}`);
-				console.error(`exec error: ${error}`);
-				done(error);
-			});
-		}
-	})
+	let vscodePath = await vscodeTest.downloadAndUnzipVSCode(vscodeVersion);
+	try {
+		await vscodeTest.runTests({
+			vscodeExecutablePath: vscodePath,
+			extensionDevelopmentPath: workspace,
+			extensionTestsPath: extensionTestsPath,
+			launchArgs: args
+		});
+	} catch (error) {
+		console.log(`stdout: ${process.stdout}`);
+		console.log(`stderr: ${process.stderr}`);
+		console.error(`exec error: ${error}`);
+	}
+	done();
+	process.exit(0);
```

Comment:
What's the exit for here? I would expect that the gulp task would exit after returning from this

Reply:
It doesn't since the vscode starts a process and the gulp task is left hanging after finishing.

Reply:
What process? The extension host process it runs tests in?

Reply:
it starts a vscode instance to run these tests so it's probably the extension host's child